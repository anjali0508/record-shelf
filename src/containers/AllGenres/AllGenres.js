/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import styles from './AllGenres.module.css';
import icon from '../../assets/images/icon-grid.svg';
import { SongCard } from '../../components/SongCard/SongCard';

export const AllGenres = ({ genreFilteredSongs, switchHeart }) => {
  const history = useHistory();
  console.log(genreFilteredSongs);
  return (
    <>
      <div className={styles.main}>
        <div className={styles.row}>
          <h1 className={styles.column_left}>genres</h1>
          <img className={styles.column_right} onClick={() => history.push('/')} src={icon} alt="toggle" />
        </div>
        <br />
        { Object.keys(genreFilteredSongs).map((category) => (
          <React.Fragment key={category}>
            <img src={`../../../images/genre-${category}.png`} alt="genre" />
            <h2>{category}</h2>
            <ul className={styles.cards}>

              {genreFilteredSongs[category].map((song) => (
                <SongCard
                  id={song.id}
                  key={song.id}
                  img={song.img}
                  name={song.name}
                  artist={song.artist}
                  likesCount={song.likes_count}
                  like={song.like}
                  switchHeart={() => switchHeart(song.id)}
                />
              ))}
            </ul>

          </React.Fragment>
        ))}
      </div>
    </>
  );
};

const Song = PropTypes.shape({
  id: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  artist: PropTypes.string.isRequired,
  likes_count: PropTypes.number.isRequired,
  like: PropTypes.bool.isRequired,
});
AllGenres.propTypes = {
  switchHeart: PropTypes.func.isRequired,
  genreFilteredSongs: PropTypes.objectOf(PropTypes.arrayOf(Song)).isRequired,
};
