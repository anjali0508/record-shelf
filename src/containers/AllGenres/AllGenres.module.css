/* Design */
* {
  box-sizing: border-box;
  font-family: "Poppins", sans-serif;
}

html {
  background-color: white;
}

.main {
  /* color: white; */
  padding: 1rem;
  max-width: 75rem;
  margin: 0 auto;
  background-color: #f5f5f5;
}

.cards {
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
}
.row {
  display: flex;
  width: 100%;
}
.column_left {
  width: 90%;
  font-size: xxx-large;
}

.column_right {
  float: right;
}

.genre_type {
  position: relative;
  padding: 2rem 1rem;
}

.blue_button {
  background-color: #4c63aa;
  border: none;
  border-radius: 30px;
  color: white;
  font-weight: 600;
  font-size: x-large;
  padding: 10px 40px;
  position: absolute;
  top: 40%;
  left: 8rem;
}
