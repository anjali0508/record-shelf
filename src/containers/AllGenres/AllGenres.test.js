import {
  render, screen, fireEvent,
} from '@testing-library/react';
import React from 'react';
import { AllGenres } from './AllGenres';

const mockData = {
  Bollywood: [
    {
      artist: 'Neha Kakkar',
      genre: 'Bollywood',
      id: '07075cdb-9e5c-41ec-8c09-9a1322dd92d8',
      img: 'https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6',
      like: true,
      likes_count: 7,
      name: 'Aankh Marey',
    },
  ],
};
describe(AllGenres.name, () => {
  test('should match snapshot', () => {
    const container = render(<AllGenres switchHeart={() => {}} genreFilteredSongs={mockData} />);
    expect(container).toMatchSnapshot();
  });
});
