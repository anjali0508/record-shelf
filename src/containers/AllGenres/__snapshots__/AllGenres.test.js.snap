// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllGenres should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main"
      >
        <div
          class="row"
        >
          <h1
            class="column_left"
          >
            genres
          </h1>
          <img
            alt="toggle"
            class="column_right"
            src="icon-grid.svg"
          />
        </div>
        <br />
        <div
          class="genre_type"
        >
          <img
            alt="genre"
            class="genre_type_img"
            src="../../../images/genre-Bollywood.png"
          />
          <button
            class="blue_button"
            type="button"
          >
            Bollywood
          </button>
        </div>
        <ul
          class="cards"
        >
          <li
            class="cards_item"
          >
            <div
              class="card"
            >
              <div
                class="card_content"
              >
                <img
                  alt="song cover"
                  class="card_image"
                  src="https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6"
                />
                <div
                  class="row"
                >
                  <div
                    class="column_left"
                  >
                    <p
                      class="card_title"
                    >
                      Aankh Marey
                    </p>
                    <p
                      class="card_text"
                    >
                      Neha Kakkar
                    </p>
                  </div>
                  <div
                    class="column_right"
                  >
                    <img
                      alt="likes"
                      class="heart_image"
                      src="heart-red.svg"
                    />
                    <p
                      class="likes_count"
                    >
                      7
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main"
    >
      <div
        class="row"
      >
        <h1
          class="column_left"
        >
          genres
        </h1>
        <img
          alt="toggle"
          class="column_right"
          src="icon-grid.svg"
        />
      </div>
      <br />
      <div
        class="genre_type"
      >
        <img
          alt="genre"
          class="genre_type_img"
          src="../../../images/genre-Bollywood.png"
        />
        <button
          class="blue_button"
          type="button"
        >
          Bollywood
        </button>
      </div>
      <ul
        class="cards"
      >
        <li
          class="cards_item"
        >
          <div
            class="card"
          >
            <div
              class="card_content"
            >
              <img
                alt="song cover"
                class="card_image"
                src="https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6"
              />
              <div
                class="row"
              >
                <div
                  class="column_left"
                >
                  <p
                    class="card_title"
                  >
                    Aankh Marey
                  </p>
                  <p
                    class="card_text"
                  >
                    Neha Kakkar
                  </p>
                </div>
                <div
                  class="column_right"
                >
                  <img
                    alt="likes"
                    class="heart_image"
                    src="heart-red.svg"
                  />
                  <p
                    class="likes_count"
                  >
                    7
                  </p>
                </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
